diff --git a/exercises/direction_processor_exercise/direction_app.py b/.details/direction_exc_ans/direction_app.py
index 75ba8c3..813accb 100755
--- a/exercises/direction_processor_exercise/direction_app.py
+++ b/.details/direction_exc_ans/direction_app.py
@@ -28,6 +28,8 @@ from pntos.cobra.config import (
     LcmLogTransportConfig,
     ManualHeadingAlignmentConfig,
     PinsonStateBlockConfig,
+    Direction3dToPointsMPConfig,
     StandardOrchestrationConfig,
     TimeAdjusterConfig,
 )
@@ -76,7 +78,23 @@ my_config = [
             label='pinson15',
             imu_model=imu_model,
         ),
-        mp_configs=(),  # <- Need something here
+        # Needed to add configuration for the MeasurementProcessor
+        mp_configs=(
+            Direction3dToPointsMPConfig(
+                group='config/direction_measurement_processor',
+                label='direction',
+                channel='/sensor/simulated/directiontoknownfeature',
+                state_block_labels=('pinson15',),
+                lever_arm=(0.80, 0.0, 0.05),
+                orientation=(0.707106781, 0.0, 0.707106781, 0.0),
+            ),
+        ),
         inertial_config=InertialConfig(
             group='config/inertial',
             expected_dt=0.01,
