diff --git a/exercises/new_app_exercise/pos_baro_app.py b/.details/new_app_exc_ans/pos_baro_app.py
index 0a0e0d9..7fffc9d 100755
--- a/exercises/new_app_exercise/pos_baro_app.py
+++ b/.details/new_app_exc_ans/pos_baro_app.py
@@ -36,6 +36,7 @@ from pntos.cobra.config import (
     AltitudeMPConfig,
     StandardOrchestrationConfig,
     TimeAdjusterConfig,
+    PinsonWithNedFogmPositionMPConfig,
 )
 from pntos_python_datasets_lcm import EXAMPLE_LCM_LOG
 
@@ -80,6 +81,20 @@ my_config = [
             imu_model=imu_model,
         ),
         additional_sb_configs=(
+            FogmStateBlockConfig(
+                group='config/pos_fogm_block',
+                label='pos_sensor_error',
+                estimate_with_covariance=EstimateWithCovariance(
+                    type=EstimateWithCovarianceType.EWC_GENERIC,
+                    estimate=np.zeros((3,)),
+                    covariance=(np.eye(3) * 9.0),
+                ),
+                fogm_model=FogmConfig(
+                    group='config/pos_sensor_error',
+                    sigma=(1.5, 1.5, 2.0),
+                    tau=(300.0, 300.0, 200.0),
+                ),
+            ),
             FogmStateBlockConfig(
                 group='config/alt_fogm_block',
                 label='alt_fogm',
@@ -96,6 +111,13 @@ my_config = [
             ),
         ),
         mp_configs=(
+            PinsonWithNedFogmPositionMPConfig(
+                group='config/pos_measurement_processor',
+                label='pos',
+                channel='/sensor/ublox-ZED-F9T/position',
+                state_block_labels=('pinson15', 'pos_sensor_error'),
+                lever_arm=(-0.50, 0.38, -0.05),
+            ),
             AltitudeMPConfig(
                 group='config/alt_measurement_processor',
                 label='alt',
