diff --git a/exercises/new_app_exercise/pos_baro_app.py b/.details/new_app_exc_ans/pos_baro_app.py
index 2ab7825..109cfc6 100644
--- a/exercises/new_app_exercise/pos_baro_app.py
+++ b/.details/new_app_exc_ans/pos_baro_app.py
@@ -23,8 +23,8 @@ from pntos.cobra import (
 )
 from pntos.cobra.config import (
     AspnVersion,
-    ControllerConfig,
     BarometerToAltitudeConfig,
+    ControllerConfig,
     FogmConfig,
     FogmStateBlockConfig,
     ImuConfig,
@@ -37,6 +37,7 @@ from pntos.cobra.config import (
     SensorMeasurementProcessorConfig,
     StandardOrchestrationConfig,
     TimeAdjusterConfig,
+    TimeBiasConfig,
 )
 from pntos_python_datasets import EXAMPLE_LCM_LOG
 
@@ -68,6 +69,7 @@ my_config = [
         channels_to_process=(
             '/sensor/vn-100/imu',
             '/sensor/ublox-ZED-F9T/position',
+            '/sensor/bmp388/baro_pressure',
         ),
     ),
     ControllerConfig(group='controller'),
@@ -81,6 +83,20 @@ my_config = [
             imu_model=imu_model,
         ),
         additional_sb_configs=(
+            FogmStateBlockConfig(
+                group='config/pos_fogm_block',
+                label='pos_sensor_error',
+                estimate_with_covariance=EstimateWithCovariance(
+                    type=EstimateWithCovarianceType.EWC_GENERIC,
+                    estimate=np.zeros((3,)),
+                    covariance=(np.eye(3) * 9.0),
+                ),
+                fogm_model=FogmConfig(
+                    group='config/pos_sensor_error',
+                    sigma=(1.5, 1.5, 2.0),
+                    tau=(300.0, 300.0, 200.0),
+                ),
+            ),
             FogmStateBlockConfig(
                 group='config/alt_fogm_block',
                 label='alt_fogm',
@@ -97,6 +113,20 @@ my_config = [
             ),
         ),
         mp_configs=(
+            SensorMeasurementProcessorConfig(
+                group='config/gps_measurement_processor',
+                identifier='pinson_with_ned_fogm_position',
+                label='gps',
+                channel='/sensor/ublox-ZED-F9T/position',
+                state_block_labels=('pinson15', 'pos_sensor_error'),
+                aux_channels=('INERTIAL_PVA',),
+                sensor_config=SensorConfig(
+                    group='config/gp3d_state_modeling',
+                    lever_arm=(-0.50, 0.38, -0.05),
+                    orientation=(0.0, 0.0, 0.0, 0.0),
+                    sensor_name='position',
+                ),
+            ),
             SensorMeasurementProcessorConfig(
                 group='config/alt_measurement_processor',
                 identifier='pinson_altitude',
@@ -137,6 +167,14 @@ my_config = [
                 channel_to_correct='/sensor/vn-100/imu',
                 expected_dt_nsec=int(0.01 * 1e9),
             ),
+            TimeBiasConfig(
+                group='config/time_bias',
+                channels_to_correct=(
+                    '/sensor/ublox-ZED-F9T/position',
+                    '/sensor/ublox-ZED-F9T/velocity',
+                ),
+                time_bias=int(0.15 * 1e9),
+            ),
             BarometerToAltitudeConfig(
                 group='config/pressure_to_alt',
                 channel='/sensor/bmp388/baro_pressure',
