diff --git a/exercises/new_app_exercise/pos_baro_app.py b/.details/new_app_exc_ans/pos_baro_app.py
index 6888c62..b939c28 100644
--- a/exercises/new_app_exercise/pos_baro_app.py
+++ b/.details/new_app_exc_ans/pos_baro_app.py
@@ -82,6 +82,20 @@ my_config = [
             imu_model=imu_model,
         ),
         additional_sb_configs=(
+            FogmStateBlockConfig(
+                group='config/pos_fogm_block',
+                label='pos_sensor_error',
+                estimate_with_covariance=EstimateWithCovariance(
+                    type=EstimateWithCovarianceType.EWC_GENERIC,
+                    estimate=np.zeros((3,)),
+                    covariance=(np.eye(3) * 9.0),
+                ),
+                fogm_model=FogmConfig(
+                    group='config/pos_sensor_error',
+                    sigma=(1.5, 1.5, 2.0),
+                    tau=(300.0, 300.0, 200.0),
+                ),
+            ),
             FogmStateBlockConfig(
                 group='config/alt_fogm_block',
                 label='alt_fogm',
@@ -97,6 +111,20 @@ my_config = [
             ),
         ),
         mp_configs=(
+            SensorMeasurementProcessorConfig(
+                group='config/gps_measurement_processor',
+                identifier='pinson_with_ned_fogm_position',
+                label='gps',
+                channel='/sensor/ublox-ZED-F9T/position',
+                state_block_labels=('pinson15', 'pos_sensor_error'),
+                aux_channels=('INERTIAL_PVA',),
+                sensor_config=SensorConfig(
+                    group='config/gp3d_state_modeling',
+                    lever_arm=(-0.50, 0.38, -0.05),
+                    orientation=(0.0, 0.0, 0.0, 0.0),
+                    sensor_name='position',
+                ),
+            ),
             SensorMeasurementProcessorConfig(
                 identifier='pinson_altitude',
                 label='alt',
