diff --git a/exercises/zupt_exercise/zupt_app.py b/.details/zupt_exc_ans/zupt_app.py
index 234a450..0babd3b 100755
--- a/exercises/zupt_exercise/zupt_app.py
+++ b/.details/zupt_exc_ans/zupt_app.py
@@ -32,6 +32,7 @@ from pntos.cobra.config import (
     InertialConfig,
     LcmLogTransportConfig,
     ManualHeadingAlignmentConfig,
+    MeasurementProcessorConfig,
     OutageConfig,
     PinsonStateBlockConfig,
     SensorConfig,
@@ -43,6 +44,8 @@ from pntos.cobra.config import (
 from pntos_python_datasets_lcm import EXAMPLE_LCM_LOG
 
 from ZuptPreprocessorPlugin import ZuptPreprocessorPlugin
+from ZuptConfig import ZuptConfig
+from numpy import array
 
 OUTPUT_LOG = sys.argv[1] if len(sys.argv) > 1 else 'pntos_output.log'
 
@@ -155,6 +158,14 @@ my_config = [
                     sensor_name='altitude',
                 ),
             ),
+            MeasurementProcessorConfig(
+                group='config/vel_measurement_processor',
+                identifier='pinson_velocity',
+                label='vel',
+                channel='/preproc/simulated/zupt',
+                state_block_labels=('pinson15',),
+                aux_channels=('INERTIAL_PVA',),
+            ),
         ),
         inertial_config=InertialConfig(
             group='config/inertial',
@@ -197,6 +208,11 @@ my_config = [
                 start_time=1000.0,
                 end_time=2700.0,
             ),
+            ZuptConfig(
+                group='config/zero_vel',
+                identifier='zupt_generator',
+                stationary_times=array(STATIONARY_TIME_RANGES, dtype=float),
+            ),
         ),
         max_prop_interval=1.0,
         group='config/orchestration',
